src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -0,0 +1,186 @@
+            }
+
+            if (comment != null)
+            {
+                update = update.Set("Comment", comment);
+            }
+
+            if (guidelineIds != null)
+            {
+                update = update.Set("GuidelineIds", guidelineIds);
+            }
+
+            if (badCode != null)
+            {
+                var embedding = await GetEmbeddingsAsync(badCode);
+                update = update
+                    .Set("BadCode", badCode)
+                    .Set("Embedding", embedding);
+            }
+
+            return await _copilotCommentsRepository.UpdateDocumentAsync(filter, update);
+        }
+
+        public async Task<UpdateResult> DeleteDocumentAsync(string user, string id)
+        {
+            var filter = GetIdFilter(id);
+
+            var updateBuilder = Builders<CopilotCommentModel>.Update;
+            var update = updateBuilder
+                .Set("IsDeleted", true)
+                .Set("ModifiedOn", DateTime.UtcNow)
+                .Set("ModifiedBy", user);
+
+            return await _copilotCommentsRepository.DeleteDocumentAsync(filter, update); 
+        }
+
+        public async Task<string> GetDocumentAsync(string id)
+        {
+            var filter = GetIdFilter(id);
+            var document = await _copilotCommentsRepository.GetDocumentAsync(filter);
+
+            var documentJson = JsonConvert.SerializeObject(document);
+
+            return documentJson;
+        }
+
+        private FilterDefinition<CopilotCommentModel> GetIdFilter(string id)
+        {
+            return Builders<CopilotCommentModel>.Filter
+                .Where(f => f.Id.ToString() == id && f.IsDeleted == false);
+        }
+
+        private async Task<float[]> GetEmbeddingsAsync(string badCode)
+        {
+            // structure of Embeddings object 
+            //{
+            //    { "Data", 
+            //        [
+            //            { "EmbeddingsItem", 
+            //                {
+            //                    { "Embedding", [ "float1", "float2" ]},
+            //                    { "Index", "1"}
+            //                }
+            //            }
+            //        ]
+            //    },
+            //    { "Usage", 
+            //        {
+            //            { "PromptTokens", "1"},
+            //            { "TotalTokens", "2"}
+            //        }
+            //    }
+            //}
+
+            var options = new EmbeddingsOptions(badCode);
+            var response = await _openAIClient.GetEmbeddingsAsync(_configuration["OpenAI:Model"], options);
+            var embeddings = response.Value;
+
+            float[] embedding = embeddings.Data[0].Embedding.ToArray();
+
+            return embedding;
+        }
+
+        public BsonDocument EmbeddingToBson(Embeddings embedding)
+        {
+            var data = embedding.Data;
+
+            BsonDocument embeddingUsageBson = new BsonDocument
```

Comment:
Either use `var` or you can use `new`, but no need to repeat yourself with `BsonDocument`.